driver-sync/src/main/com/mongodb/client/internal/MongoBatchCursorAdapter.java

```diff
@@ -96,6 +99,27 @@ private T getNextInBatch() {
             curBatch = null;
             curPos = 0;
         }
+
+        setResumeToken(nextInBatch);
         return nextInBatch;
     }
+
+    private void setResumeToken(final T nextInBatch) {
+        if (nextInBatch instanceof ChangeStreamDocument) {
+            resumeToken = ((ChangeStreamDocument) nextInBatch).getResumeToken();
```

Comment:
If users aren't using `ChangeStreamDocument` with change streams then the resumeToken won't be updated.

Reply:
Looking at the spec, `ChangeStreamDocument` is the only structure that contains the resume token. If `nextInBatch` is any other structure, there is no access to a resume token that I am aware of.

Reply:
The issue is if the user wants the change stream documents as BsonDocuments like so:

`mongoCollection.watch().withDocumentClass(BsonDocument.class).iterable()` 

The the resume token won't be saved as they iterate.

Reply:
I have removed this method and emulated the C# implementation to set the resume token to the token found in the last document in the batch in `ChangeStreamBatchCursor#convertResults`.